Asserts: Add expectThrowable() method

burned42 opened the pull request:
I thought it would be quite useful to be also able to test for [Errors](http://php.net/manual/de/class.error.php) and (possibly in future) other [Throwables](http://php.net/manual/de/class.throwable.php), too instead of just [Exceptions](http://php.net/manual/de/class.exception.php).

So this pull request would add a new method `expectThrowable()` and refactor `expectException()` to just call this method. I also suggest marking `expectException()` as deprecated since I think the old method could be removed some time later (maybe with a new major release or such?). I also added some tests and updated the documentation. Please tell me what you think about this and if there are any additional steps required that I didn't know of.

DavertMik commented on tests/unit/Codeception/Module/AssertsTest.php:
```diff
@@ -66,8 +71,60 @@ public function testExceptionFails()
+    }
+
+    /**
+     * @expectedException PHPUnit\Framework\AssertionFailedError
```
Hey, could you update the code to not use this annotation? 
Because of https://github.com/sebastianbergmann/phpunit/issues/3332

burned42 replied:
Yes, sure, thanks for pointing that out, didn't know that! :)

Fixed in f8a2b5289cc907bb095ee0be889d1fa86ef086af

DavertMik commented on tests/unit/Codeception/Module/AssertsTest.php:
```diff
@@ -66,8 +71,60 @@ public function testExceptionFails()
+    }
+
+    /**
+     * @expectedException PHPUnit\Framework\AssertionFailedError
```
Please don't use annotations https://github.com/sebastianbergmann/phpunit/issues/3332

burned42 replied:
Fixed in f8a2b5289cc907bb095ee0be889d1fa86ef086af

DavertMik commented on tests/unit/Codeception/Module/AssertsTest.php:
```diff
@@ -66,8 +71,60 @@ public function testExceptionFails()
+    }
+
+    /**
+     * @expectedException PHPUnit\Framework\AssertionFailedError
```
Please don't use annotations https://github.com/sebastianbergmann/phpunit/issues/3332

burned42 replied:
Fixed in f8a2b5289cc907bb095ee0be889d1fa86ef086af

DavertMik commented on tests/unit/Codeception/Module/AssertsTest.php:
```diff
@@ -66,8 +71,60 @@ public function testExceptionFails()
+
+    /**
+     * @expectedException PHPUnit\Framework\AssertionFailedError
+     * @expectedExceptionMessageRegExp /RuntimeException/
```
Please don't use annotations https://github.com/sebastianbergmann/phpunit/issues/3332

burned42 replied:
Fixed in f8a2b5289cc907bb095ee0be889d1fa86ef086af

burned42 commented:
> > I also suggest marking expectException() as deprecated since I think the old method could be removed some time later
> 
> I think it's ok how it is. I got used to it )

Alright, so should I then just remove those lines?
https://github.com/Codeception/Codeception/pull/5213/files#diff-214a6d3c36d25915108ad90033f96177R319
https://github.com/Codeception/Codeception/pull/5213/files#diff-b3d3622a11b4688b45ee6999ce435bc0R459 ?

> 
> > I also added some tests and updated the documentation. Please tell me what you think about this and if there are any additional steps required that I didn't know of.
> 
> Thanks. Everything looks fine, I just didn't have enough time to review it before

Thanks for reviewing :)

DavertMik commented:
Thanks. Let's do that!
